File: android/src/main/java/com/svgaplayer/events/TopFrameEvent.java
package com.svgaplayer.events;

import androidx.annotation.Nullable;

import com.facebook.react.bridge.WritableMap;
import com.facebook.react.uimanager.events.Event;

public class TopFrameEvent extends Event<TopFrameEvent> {
    public static String  EVENT_NAME = "onFrame";
    private WritableMap eventData;
    public TopFrameEvent(int surfaceId, int viewTag, WritableMap data) {
        super(surfaceId, viewTag);
        eventData = data;
    }

    @Override
    public String getEventName() {
        return EVENT_NAME;
    }

    @Nullable
    @Override
    protected WritableMap getEventData() {
        return eventData;
    }
}

File: android/src/main/java/com/svgaplayer/events/TopFrameEvent.kt
package com.svgaplayer.events

import com.facebook.react.bridge.WritableMap
import com.facebook.react.uimanager.events.Event

class TopFrameEvent(
    surfaceId: Int,
    viewTag: Int,
    private val eventData: WritableMap
) : Event<TopFrameEvent>(surfaceId, viewTag) {

    companion object {
        const val EVENT_NAME = "onFrameChanged"
    }

    override fun getEventName(): String = EVENT_NAME

    override fun getEventData(): WritableMap? = eventData
}
